widget/calendar.go

```diff
@@ -0,0 +1,229 @@
+	dayIndex := int(start.Weekday())
+	if dayIndex == 0 {
+		dayIndex += 7
+	}
+
+	//add spacers if week doesn't start on Monday
+	for i := 0; i < dayIndex-1; i++ {
+		buttons = append(buttons, layout.NewSpacer())
+	}
+
+	for d := start; d.Month() == start.Month(); d = d.AddDate(0, 0, 1) {
+
+		dayNum := d.Day()
+		s := strconv.Itoa(dayNum)
+		var b fyne.CanvasObject = widget.NewButton(s, func() {
+
+			selectedDate := c.dateForButton(dayNum)
+
+			c.onSelected(selectedDate)
+
+			c.hideOverlay()
+		})
+
+		buttons = append(buttons, b)
+	}
+
+	return buttons
+}
+
+func (c *Calendar) dateForButton(dayNum int) time.Time {
+	oldName, off := c.calendarTime.Zone()
+	return time.Date(c.year, time.Month(c.month), dayNum, c.calendarTime.Hour(), c.calendarTime.Minute(), 0, 0, time.FixedZone(oldName, off)).In(c.calendarTime.Location())
+}
+
+func (c *Calendar) hideOverlay() {
+	overlayList := c.canvas.Overlays().List()
+	overlayList[0].Hide()
+}
+
+func (c *Calendar) monthYear() string {
+	return time.Month(c.month).String() + " " + strconv.Itoa(c.year)
+}
+
+func (c *Calendar) calendarObjects() []fyne.CanvasObject {
+	textSize := float32(8)
+	columnHeadings := []fyne.CanvasObject{}
+	for i := 0; i < 7; i++ {
+		j := i + 1
+		if j == 7 {
+			j = 0
+		}
+
+		var canvasObject fyne.CanvasObject = canvas.NewText(strings.ToUpper(time.Weekday(j).String()[:3]), color.NRGBA{0xFF, 0xFF, 0xFF, 0xBF})
+		canvasObject.(*canvas.Text).TextSize = textSize
+		canvasObject.(*canvas.Text).Alignment = fyne.TextAlignCenter
+		columnHeadings = append(columnHeadings, canvasObject)
+	}
+	columnHeadings = append(columnHeadings, c.daysOfMonth()...)
+
+	return columnHeadings
+}
+
+// CreateRenderer returns a new WidgetRenderer for this widget.
+// This should not be called by regular code, it is used internally to render a widget.
+func (c *Calendar) CreateRenderer() fyne.WidgetRenderer {
+	c.monthPrevious = widget.NewButtonWithIcon("", theme.NavigateBackIcon(), func() {
+		c.month--
+		if c.month < 1 {
+			c.month = 12
+			c.year--
+		}
+		c.monthLabel.ParseMarkdown(c.monthYear())
+
+		c.dates.Objects = c.calendarObjects()
+	})
+	c.monthNext = widget.NewButtonWithIcon("", theme.NavigateNextIcon(), func() {
+		c.month++
+		if c.month > 12 {
+			c.month = 1
+			c.year++
+		}
+		c.monthLabel.ParseMarkdown(c.monthYear())
+
+		c.dates.Objects = c.calendarObjects()
+	})
+
+	c.monthLabel = widget.NewRichTextFromMarkdown(c.monthYear())
+
+	nav := container.New(layout.NewBorderLayout(nil, nil, c.monthPrevious, c.monthNext),
+		c.monthPrevious, c.monthNext, container.NewCenter(c.monthLabel))
+
+	c.dates = container.New(newCalendarLayout(c.cellSize, c.padding), c.calendarObjects()...)
+
+	dateContainer := container.NewVBox(nav, c.dates)
+
+	return widget.NewSimpleRenderer(dateContainer)
+}
+
+// NewCalendar creates a calendar instance
+func NewCalendar(cT time.Time, onSelected func(time.Time), cellSize float32, padding float32) *Calendar {
```

Comment:
It would be better if the `cellSize` just could be computed out of the size of the widget. That's more in line with how we do sizing in Fyne.

I'd also prefer if the padding could use `theme.Padding()` from Fyne instead.

Reply:
The problem with using `theme.padding()` is the date buttons are too spaced out if you want any padding at all in the app. In fact, my examples had the calendar padding at 0. I think I will remove the padding for the moment.

Reply:
If we want it to be packed with no padding that is cool - but I think omit the `padding` parameter as @Jacalz is right that's not very Fyne.